packages/algob/src/builtin-tasks/compile.ts

```diff
@@ -1,13 +1,112 @@
+import type { Algodv2, CompileOut } from "algosdk";
+import { readdirSync, readFileSync, writeFileSync } from 'fs';
+import * as path from 'path';
+import YAML from "yaml";
+
 import { task } from "../internal/core/config/config-env";
-// import { createClient } from "../lib/driver";
-// import { AlgobRuntimeEnv, TaskArguments } from "../types";
+import { parseAlgorandError } from "../internal/core/errors";
+import { assertDir, ASSETS_DIR, CACHE_DIR } from "../internal/core/project-structure";
+import { createClient } from "../lib/driver";
+import type { AlgobRuntimeEnv, ASCCache, Network } from "../types";
 import { TASK_COMPILE } from "./task-names";
+const murmurhash = require('murmurhash'); // eslint-disable-line @typescript-eslint/no-var-requires
+
+const tealExt = ".teal";
 
 export default function (): void {
-  task(TASK_COMPILE, "Compilation task")
-    .setAction(
-      async () => {
-        console.log("compiling......");
-      }
-    );
+  task(TASK_COMPILE, "Compile all TEAL smart contracts")
+    .addFlag("force", "recompile even if the source file didn't change")
+    .setAction(_compile);
```

Comment:
Let's not have both `_compile` and `compile` if they do different things and operate in different contexts.
I'd prefer `runCompile` for this one.

Reply:
how about `compileCommand` and `compile`? `do` vs `run` is not very informative

Reply:
Looks fine.